model/inline/src/main/scala/aqua/model/inline/raw/ApplyStreamMapRawInliner.scala

```diff
@@ -0,0 +1,364 @@
+    } yield {
+      val getKeysTree = getKeysStreamModel(
+        mapName = mapName,
+        mapType = mapType,
+        streamVar = streamVar,
+        iterName = iterName
+      )
+
+      val inline = Inline(predo = Chain.one(getKeysTree))
+
+      (streamVar, inline)
+    }
+  }
+
+  private def getStream[S: Mangler: Exports: Arrows: Config](
+    mapName: String,
+    mapType: StreamMapType,
+    idxVar: ValueModel
+  ): State[S, (VarModel, Inline)] = {
+    for {
+      streamName <- mang(mapName, "get_stream_result")
+      iterName <- mang(mapName, "iter")
+      value = VarModel(
+        streamName,
+        StreamType(mapType.element)
+      )
+      // add resulted stream to restrict it after inlining
+      _ <- Exports[S].resolved(streamName, value)
+    } yield {
+      val getStreamResultTree = getStreamModel(
+        mapName = mapName,
+        mapType = mapType,
+        idxVar = idxVar,
+        streamVar = value,
+        iterName = iterName
+      )
+
+      val inline = Inline(predo = Chain.one(getStreamResultTree))
+
+      (value, inline)
+    }
+  }
+
+  private def get[S: Mangler: Exports: Arrows: Config](
+    mapName: String,
+    mapType: StreamMapType,
+    idxVar: ValueModel
+  ): State[S, (VarModel, Inline)] = {
+    for {
+      canonMapName <- mang(mapName, "canon")
+      idxName <- mang(mapName, "idx")
+    } yield {
+      val (result, getResultTree) = getModel(
+        mapName,
+        mapType,
+        idxVar,
+        canonMapName,
+        idxName
+      )
+
+      val inline = Inline(predo = Chain.one(getResultTree))
+
+      (result, inline)
+    }
+  }
+
+  private def flatProps[S: Mangler](
+    arg: ValueModel,
+    op: ValueModel => State[S, (VarModel, Inline)]
+  ): State[S, (VarModel, Inline)] =
+    (arg match {
+      case vm @ VarModel(name, _, properties) if properties.nonEmpty =>
+        mang(name, "ap").map { n =>
+          VarModel(n, vm.`type`) -> Some(FlattenModel(vm, n).leaf)
+        }
+      case a => State.pure(a -> None)
+    }).flatMap { case (v, inl) =>
+      op(v).map { case (res, resInl) =>
+        res -> resInl.prepend(inl)
+      }
+    }
+
+  def apply[S: Mangler: Exports: Arrows: Config](
+    funcName: String,
+    mapName: String,
+    mapType: StreamMapType,
+    args: List[ValueModel]
+  ): State[S, (VarModel, Inline)] =
+    StreamMapType
+      .funcByString(funcName)
+      .tupleRight(args)
+      .map {
+        // flat properties for Get and Contains, because argument uses in IntoIndexModel
+        case (Get, arg :: Nil) =>
+          flatProps(arg, get(mapName, mapType, _))
+        case (Contains, arg :: Nil) =>
+          flatProps(arg, contains(mapName, mapType, _))
+        case (GetStream, arg :: Nil) =>
+          getStream(mapName, mapType, arg)
+        case (Keys, Nil) =>
+          getKeys(mapName, mapType)
+        case (KeysStream, Nil) =>
+          getKeysStream(mapName, mapType)
+        case (n, args) =>
```

Comment:
Why not put this argument flattening inside `getModel`? It already flattens the id if it is a string literal. And `contains` uses `getModel` inside, so there will be no need to use it twice.